src/components/orders/FilledProgress/index.tsx

```diff
@@ -26,13 +33,71 @@ const Wrapper = styled.div`
 `
 
 export function FilledProgress(props: Props): JSX.Element {
-  const { percentage = '0' } = props
+  const {
+    order: {
+      filledAmount,
+      filledPercentage,
+      kind,
+      buyAmount,
+      sellAmount,
+      buyToken,
+      sellToken,
+      buyTokenAddress,
+      sellTokenAddress,
+    },
+  } = props
+
+  let mainToken
+  let mainAddress
+  let mainAmount
+  let swappedToken
+  let swappedAddress
+  let swappedAmount
+  let action
+
+  if (kind === 'sell') {
+    mainToken = sellToken
+    mainAddress = sellTokenAddress
+    mainAmount = sellAmount
+    swappedToken = buyToken
+    swappedAddress = buyTokenAddress
+    swappedAmount = buyAmount
+    action = 'sold'
+  } else {
+    mainToken = buyToken
+    mainAddress = buyTokenAddress
+    mainAmount = buyAmount
+    swappedToken = sellToken
+    swappedAddress = sellTokenAddress
+    swappedAmount = sellAmount
+    action = 'bought'
+  }
+
+  // In case the token object is empty, display the address
+  const mainSymbol = mainToken ? safeTokenName(mainToken) : mainAddress
+  const swappedSymbol = swappedToken ? safeTokenName(swappedToken) : swappedAddress
+  // In case the token object is empty, display the raw amount (`decimals || 0` part)
+  const formattedFilledAmount = formatSmart(filledAmount.toString(10), mainToken?.decimals || 0)
+  const formattedMainAmount = formatSmart(mainAmount.toString(10), mainToken?.decimals || 0)
+  const formattedSwappedAmount = formatSmart(swappedAmount.toString(10), swappedToken?.decimals || 0)
+
+  const formattedPercentage = filledPercentage.times('100').toString(10)
 
   return (
     <Wrapper>
-      <ProgressBar percentage={percentage} />
+      <ProgressBar percentage={formattedPercentage} />
       <span>
-        <b>2,430 DAI</b> of <b>3000 DAI</b> sold for a total of <b>2.842739643 ETH</b>
+        <b>
+          {formattedFilledAmount} {mainSymbol}
+        </b>{' '}
+        of{' '}
+        <b>
+          {formattedMainAmount} {mainSymbol}
+        </b>{' '}
```

Comment:
Question for the reviewer: should we always show both amounts, or only when not 100% filled?
![screenshot_2021-03-03_11-08-31](https://user-images.githubusercontent.com/43217/109858533-cd436880-7c10-11eb-83b2-14f376a5631a.png)

Reply:
I'd say showing both amounts feels more consistent. Although I could see only showing the full amount when it's fully filled.